src/clu_main.c

```diff
@@ -328,3 +328,63 @@ int main(int argc, char** argv)
     return (ret == WOLFCLU_SUCCESS)? 0 : ret;
 }
 
+#ifdef WOLFCLU_FREERTOS
+
+#ifndef HAL_CONSOLE_UART
+#define HAL_CONSOLE_UART huart4
+#endif
+extern UART_HandleTypeDef HAL_CONSOLE_UART;
+
+/* When building on FreeRTOS, clu_entry() acts as the entry function and
+ * parses the UART-transmitted command. Be sure to rename the main function
+ *  above, to clu_main() so the right function is called below */
+int clu_entry(const void* argument)
+{
+
+	HAL_StatusTypeDef halRet;
+	uint8_t buffer[50];
+
+	WOLFCLU_LOG(WOLFCLU_L0, "Please enter a wolfCLU command (wolfssl -h for help)");
+
+        /* Recieve the command from the UART console */
+        do {
+	    halRet = HAL_UART_Receive(&HAL_CONSOLE_UART, buffer, sizeof(buffer),100);
+	} while (halRet != HAL_OK || buffer[0] == '\n' || buffer[0] == '\r');
+
+	WOLFCLU_LOG(WOLFCLU_L0, "Command received.");
+
+	char* command = (char*)buffer;
+	char* token;
+	int argc = 1;
+	int i = 0;
+
+	for(i=0;command[i]!='\0';i++){
+		if (command[i]==' '){
+			argc++;
+		}
+	}
+
+	char* argv[argc];
```

Comment:
variable declaration should be at top of scope

Reply:
This specific one can't be moved because the size of the array as calculated immediately above

Reply:
that brings up a different issue that I missed on first review, lets avoid declaring a variable size array. It's not great for portability / embedded cases

Reply:
I agree, but the problem I saw was that we do not know how many arguments a user will enter as they'll differ by command

Reply:
it can be created dynamically with a malloc call in that case or have a max value like "#define MAX_COMMAND_ARGS 30" and error check if more than expected. If doing a max command arg then I suggest having it be like

```
#ifndef MAX_COMMAND_ARGS
#define MAX_COMMAND_ARGS 30
#endif
```

when like that it can be overridden by setting the MAX_COMMAND_ARGS at compile time. Either route would work

Reply:
That sounds good.

Reply:
With using MAX_COMMAND_ARGS lets add a check that the index into argv does not go past the max size.

Reply:
Done